contracts/OperatorStage2.sol

```diff
@@ -48,12 +65,33 @@ contract OperatorStage2 is AccessControl {
 		tokenGateway = StandardTokenGateway(gatewayAddress);
 	}
 
+	// Adds a new rate that allows a user to bond with
+	function addRate(uint256 _rate, uint256 _maturityInWeeks) public onlyOperatorStage2 {
+		allowedYieldToWeeks[_rate] = _maturityInWeeks;
+		BondRate memory br = BondRate(_rate, _maturityInWeeks);
+		ratesAvailable.push(br);
+		emit Yield(_rate, _maturityInWeeks);
+	}
+
+	// Sets a rate to zero, not removing it but making it obsolete
+	// TODO: make this nicer and not just bond for 0% yield
+	function removeRate(uint256 _rate) public onlyOperatorStage2 {
+		allowedYieldToWeeks[_rate] = 0;
+		emit Yield(_rate, 0);
```

Comment:
can we just delete `delete allowedYieldToWeeks[_rate]` ?

Reply:
as far as I remember, it's not that simple to delete it directly. rather we can loop on the frontend to see which keys have a rate of non-zero instead. don't expect that we have that many interest rates anyway.

since default (and uninitialized) values of maps are zero, setting them to zero will have the same effect here.

you can try a simple example of deleting element in map and then check the size to see if it is reduced in the unit tests.

Reply:
you are quite correct sir 🙂